internal/ingress/metric/main.go

```diff
@@ -157,16 +158,39 @@ func (c *collector) SetSSLExpireTime(servers []*ingress.Server) {
 }
 
 func (c *collector) SetHosts(hosts sets.String) {
+	if !isLeader() {
+		return
+	}
+
```

Comment:
Why?

Reply:
Only the current leader should update this metrics (to avoid duplication)
https://github.com/kubernetes/ingress-nginx/pull/3885

Reply:
fixed